packages/react/src/components/FormattingToolbar/DefaultButtons/ColorStyleButton.tsx

```
@@ -1,64 +1,124 @@
+        };
+      }
+> {
+  return (
+    `${color}Color` in editor.styleSchema &&
+    editor.styleSchema[`${color}Color`].type === `${color}Color` &&
+    editor.styleSchema[`${color}Color`].propSchema === "string"
+  );
+}
 
-export const ColorStyleButton = <BSchema extends BlockSchema>(props: {
-  editor: BlockNoteEditor<BSchema>;
-}) => {
-  const selectedBlocks = useSelectedBlocks(props.editor);
+export const ColorStyleButton = () => {
+  const editor = useBlockNoteEditor<
+    BlockSchema,
+    InlineContentSchema,
+    StyleSchema
+  >();
+
+  const textColorInSchema = checkColorInSchema("text", editor);
+  const backgroundColorInSchema = checkColorInSchema("background", editor);
+
+  const selectedBlocks = useSelectedBlocks(editor);
 
   const [currentTextColor, setCurrentTextColor] = useState<string>(
-    props.editor.getActiveStyles().textColor || "default"
+    textColorInSchema
+      ? editor.getActiveStyles().textColor || "default"
+      : "default"
   );
   const [currentBackgroundColor, setCurrentBackgroundColor] = useState<string>(
-    props.editor.getActiveStyles().backgroundColor || "default"
+    backgroundColorInSchema
+      ? editor.getActiveStyles().backgroundColor || "default"
+      : "default"
   );
 
   useEditorContentOrSelectionChange(() => {
-    setCurrentTextColor(props.editor.getActiveStyles().textColor || "default");
-    setCurrentBackgroundColor(
-      props.editor.getActiveStyles().backgroundColor || "default"
-    );
-  }, props.editor);
+    if (textColorInSchema) {
+      setCurrentTextColor(editor.getActiveStyles().textColor || "default");
+    }
+    if (backgroundColorInSchema) {
+      setCurrentBackgroundColor(
+        editor.getActiveStyles().backgroundColor || "default"
+      );
+    }
+  }, editor);
 
   const { ref, updateMaxHeight } = usePreventMenuOverflow();
 
   const setTextColor = useCallback(
     (color: string) => {
-      props.editor.focus();
+      if (!textColorInSchema) {
+        return;
```

can this happen? or should we throw an error?